StickKnightsOnline-Client/main.cpp

```diff
@@ -8204,12 +8253,53 @@ void physics()
 
 
 
+	  //Sign reading collision detection
+	  if (enableSIGN) {
+		  for (int i = 0; i < map[current_map]->num_signs; i++)
+		  {
+			  float rangeX = (std::abs)(Player[MyID].x - map[current_map]->Sign[i].x);
+			  float rangeY = (std::abs)(Player[MyID].y - map[current_map]->Sign[i].y);
+			  float distance = sqrt((rangeX*rangeX) + (rangeY*rangeY));
+			  bool inRange;
+			  //isColliding = blocked(Player[MyID].x, Player[MyID].y, map[current_map]->Sign[i].x, map[current_map]->Sign[i].y);
+
+			  inRange = (distance < 50);
+			  if (inRange) {
+				  printf("sign!\n");
+				  bool isNew = ((current_sign.hasBeenClosed) == (map[current_map]->Sign[i].hasBeenClosed));
+				  if (isNew){
+					  current_sign = map[current_map]->Sign[i];
+				  }
+				  if (!current_sign.triggered && !(current_sign.hasBeenClosed)) {
+					  current_sign.triggered = true;
+					  popup_menu = 0;
+					  popup_sign = true;
+					  popup_npc = false;
+				  }
+			  }
 
+			  /*if ( (distance > 175) && (map[current_map]->Sign[i].triggered == true)) {
+				  popup_sign = false;
+			  }*/
 
 
 
+		  }
+	  
+		  float rangeX = (std::abs)(Player[MyID].x - current_sign.x);
```

Comment:
Possibly could use this math library to simplify some of the code in other areas of the program. I don't mind including it if it's useful and easy to read.